sql/src/main/java/io/crate/analyze/OrderBy.java

```diff
@@ -53,6 +69,55 @@ public OrderBy(List<Symbol> orderBySymbols, boolean[] reverseFlags, Boolean[] nu
         this.nullsFirst = nullsFirst;
     }
 
+    /**
+     * Creates a new OrderBy with the other symbols prepended (or returns this if symbols are empty).
+     * Symbols are de-duplicated to some degree, e.g.
+     *
+     * <pre>
+     *     ORDER BY x y `prependUnique` x z will result in x z y; without duplicating x
+     * </pre>
+     *
+     * The defaults for reverseFlags and nullsFirst are used (asc, undefined)
+     */
+    public OrderBy prependUnique(Collection<? extends Symbol> symbols) {
+        if (symbols.isEmpty()) {
+            return this;
+        }
+        ArrayList<Symbol> newOrderBySymbols = new ArrayList<>();
+        ArrayList<Boolean> newReverseFlags = new ArrayList<>();
+        ArrayList<Boolean> newNullsFirst = new ArrayList<>();
+        var orderBySymbols = this.orderBySymbols.listIterator();
+        var xsToPrepend = symbols.iterator();
+        var nextOrderBy = orderBySymbols.hasNext() ? orderBySymbols.next() : null;
+        while (xsToPrepend.hasNext()) {
+            Symbol toPrepend = xsToPrepend.next();
+            if (nextOrderBy == null && orderBySymbols.hasNext()) {
+                nextOrderBy = orderBySymbols.next();
+            }
+            if (toPrepend.equals(nextOrderBy)) {
+                newOrderBySymbols.add(nextOrderBy);
+                newReverseFlags.add(reverseFlags[orderBySymbols.previousIndex()]);
+                newNullsFirst.add(nullsFirst[orderBySymbols.previousIndex()]);
+                nextOrderBy = null;
```

Comment:
on second thought, this null "flagging" is a bit misleading and hard to follow. should we set the `nextOrderBy` value to the next value here instead of above? 

```
if (orderBySymbols.hasNext()) {
    nextOrderBy = orderBySymbols.next();
} else {
    nextOrderBy = null;
}

Reply:
@mfussenegger what do you think of this ^ ?

Reply:
@andreidan pushed a fixup